eng/pipelines/templates/jobs/smoke.tests.yml

```diff
@@ -34,89 +33,98 @@ jobs:
+          Python_38_Linux (AzureCloud Canary):
+            PythonVersion: '3.8'
+            Pool: "azsdk-pool-mms-ubuntu-2004-general"
+            OSVmImage: "MMSUbuntu20.04"
+            SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources-preview)
+            ArmTemplateParameters: $(azureCloudArmParameters)
+            Location: 'eastus2euap'
+          Python_37_Windows (AzureCloud):
+            PythonVersion: '3.7'
+            Pool: "azsdk-pool-mms-win-2019-general"
+            OSVmImage: "MMS2019"
+            SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources)
+            ArmTemplateParameters: $(azureCloudArmParameters)
+          Python_38_Windows (AzureCloud):
+            PythonVersion: '3.8'
+            Pool: "azsdk-pool-mms-ubuntu-2004-general"
+            OSVmImage: "MMSUbuntu20.04"
+            SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources)
+            ArmTemplateParameters: $(azureCloudArmParameters)
+          Python_37_Mac (AzureCloud):
+            PythonVersion: '3.7'
+            Pool: Azure Pipelines
+            OSVmImage: macOS-10.15
+            SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources)
+            ArmTemplateParameters: $(azureCloudArmParameters)
+          Python_38_Mac (AzureCloud):
+            PythonVersion: '3.8'
+            Pool: Azure Pipelines
+            OSVmImage: macOS-10.15
+            SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources)
+            ArmTemplateParameters: $(azureCloudArmParameters)
+          Python_38_Linux (AzureUSGovernment):
+            PythonVersion: '3.8'
+            Pool: "azsdk-pool-mms-ubuntu-2004-general"
+            OSVmImage: "MMSUbuntu120.04"
+            SubscriptionConfiguration: $(sub-config-gov-test-resources)
+            ArmTemplateParameters: $(azureUSGovernmentArmParameters)
+          Python_37_Windows (AzureUSGovernment):
+            PythonVersion: '3.7'
+            Pool: "azsdk-pool-mms-ubuntu-2004-general"
+            OSVmImage: "MMSUbuntu20.04"
+            SubscriptionConfiguration: $(sub-config-gov-test-resources)
+            ArmTemplateParameters: $(azureUSGovernmentArmParameters)
+          Python_38_Linux (AzureChinaCloud):
+            PythonVersion: '3.8'
+            Pool: "azsdk-pool-mms-ubuntu-2004-general"
+            OSVmImage: "MMSUbuntu20.04"
+            SubscriptionConfiguration: $(sub-config-cn-test-resources)
+            Location: 'chinanorth'
+            ArmTemplateParameters: $(azureChinaCloudArmParameters)
+          Python_37_Windows (AzureChinaCloud):
+            PythonVersion: '3.7'
+            Pool: "azsdk-pool-mms-ubuntu-2004-general"
+            OSVmImage: "MMSUbuntu20.04"
+            SubscriptionConfiguration: $(sub-config-cn-test-resources)
+            Location: 'chinanorth'
+            ArmTemplateParameters: $(azureChinaCloudArmParameters)
+      ${{ if eq(parameters.Daily, false) }}:
+        matrix:
```

Comment:
I think this could be a little simpler if you keep the original matrix definition and use the conditional for the entries, like:

```
strategy:
  matrix:
    Python_38_Linux (AzureCloud):
      PythonVersion: '3.8'
      Pool: "azsdk-pool-mms-ubuntu-2004-general"
      OSVmImage: "MMSUbuntu20.04"
      SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources)
      ArmTemplateParameters: $(azureCloudArmParameters)
    ${{ if eq(parameters.Daily, true) }}:
      Python_27_Linux (AzureCloud):
        PythonVersion: '2.7'
        SkipAsyncInstall: true
        Pool: "azsdk-pool-mms-ubuntu-2004-general"
        OSVmImage: "MMSUbuntu20.04"
        SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources)
        ArmTemplateParameters: $(azureCloudArmParameters)

        ... remaining entries
```

The downside is the test ordering will be slightly less logical, but I think that's ok.